cmd/create/cluster/cmd.go

```diff
@@ -418,40 +399,98 @@ func run(cmd *cobra.Command, _ []string) {
 
 	// AWS ARN Role
 	roleARN := args.roleARN
-	if roleARN != "" && interactive.Enabled() {
+
+	if !interactive.Enabled() && creator.IsSTS && roleARN == "" {
+		err = interactive.PrintHelp(interactive.Help{
+			Message: "Since your AWS credentials are returning an STS ARN you can only " +
+				"create STS clusters. Otherwise, switch to IAM credentials.",
+		})
+		if err != nil {
+			reporter.Errorf("%s", err)
+			os.Exit(1)
+		}
+		interactive.Enable()
+		reporter.Infof("Enabling interactive mode")
+	}
+
+	if interactive.Enabled() {
 		roleARN, err = interactive.GetString(interactive.Input{
 			Question: "Role ARN",
 			Help:     cmd.Flags().Lookup("role-arn").Usage,
 			Default:  roleARN,
+			Required: creator.IsSTS,
 		})
 		if err != nil {
 			reporter.Errorf("Expected a valid ARN: %s", err)
 			os.Exit(1)
 		}
 	}
+	if roleARN != "" {
+		_, err = arn.Parse(roleARN)
```

Comment:
For a subsequent PR but I noticed this doesn't 100% parse/validate as you or I would expect, ie this ARN passes `arn:aws:sts::"123456789123":assumed-role/OrganizationAccountAccessRole/SREAccess` :disappointed: fine for now.